airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -103,3 +86,35 @@ export const useConnectorBuilderState = (): Context => {
 
   return connectorBuilderState;
 };
+
+export const useSelectedPageAndSlice = () => {
```

Comment:
This is a new hook separated from the rest, because it is only used in one component (ResultDisplay) to manage the selected page and slice per stream, and that component is only rendered if there is a selected stream.